src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1412,7 +1412,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		}
 
 		// First onExit to consumers, this can happen after the terminal has already been disposed.
-		this._onExit.fire(this._exitCode);
+		this._onExit.fire(parsedExitResult);
```

Comment:
The introduction of `null` seems pretty confusing, could this instead fire `exitCodeOrError` which would let you check `message` directly? If `exitCodeOrError` is an `ITerminalLaunchError` then the terminal failed before the process launched which is the case you're after.

Reply:
The `message` is localized, so I'm not sure what I should check for there. 

Checking if `exitCodeOrError` is an `ITerminalLaunchError` will work though. I didn't realize it would only be an `ITerminalLaunchError` when the terminal failed before the process launched.

Reply:
`parsedExitResult` is localized for the notification, `exitCodeOrError.message` isn't though.